### PR TITLE
build: add shellcheck to travis run

### PR DESCRIPTION
**Pull request by commodo**
Since there are a few shell scripts in mstpd, makes sense to try to automate any stuff that would come up later in the review process.

shellcheck looks like a good tool.
it does not enforce style, but rather correctness & compatibility with various shells

let's see how this goes

@PaulSD can you review please ?
of course, anyone else is also welcome to review

### CODE REVIEW

**Comment by commodo** [comment]
no worries

File: utils/mstpctl-utils-functions.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@ mstpctl_parse_ports()
 {
   while [ x"${1+set}" = xset ]
   do
-    # For compatibility: the `all' option.
+    # For compatibility: the 'all' option.
```
Comment (PaulSD) [inline utils/mstpctl-utils-functions.sh]: utils/mstpctl-utils-function.sh is copied almost verbatim from /lib/bridge-utils/bridge-utils.sh in the Debian bridge-utils package.  These changes look good to me, but just be aware that applying these changes means we are forking the development of this file.

File: utils/ifupdown.sh.in [hunk 2]
```diff
@@ -46,6 +46,7 @@ elif [ "$MODE" = 'stop' ]; then
   "$ip" link set dev "$IFACE" down || exit 2
 fi
 
+# shellcheck disable=SC2086
```
Comment (PaulSD) [inline utils/ifupdown.sh.in]: You removed the all_interfaces= && line in a later commit.  Does this disable line specifically apply to the all_interfaces= line, or does it apply to other lines as well?

File: utils/mstpctl-utils-functions.sh [hunk 3]
```diff
@@ -1,3 +1,4 @@
+#!/bin/sh
```
Comment (PaulSD) [inline utils/mstpctl-utils-functions.sh]: Should probably have a blank line after the shebang and before mstpctl_parse_ports() ?

File: utils/mstpctl-utils-functions.sh [hunk 4]
```diff
@@ -6,6 +6,7 @@ mstpctl_parse_ports()
     # For compatibility: the 'all' option.
     case $1 in
       all)
+	# shellcheck disable=SC2121
 	shift &&
 	set regex eth.\* em.\* 'p[0-9].*' noregex "$@"
```
Comment (PaulSD) [inline utils/mstpctl-utils-functions.sh]: `set` is being used here to replace the positional parameters (to replace $@ with some new values).  Usually there should be two dashes after `set` when using it to replace the positional parameters: `set -- regex ...`
I haven't tested, but it looks like shellcheck will not warn about this if there are two dashes after set, so maybe try adding those?

File: utils/mstpctl-utils-functions.sh (hunk 3 above)
Comment (commodo) [inline utils/mstpctl-utils-functions.sh]: ack

File: utils/mstpctl-utils-functions.sh (hunk 4 above)
Comment (commodo) [inline utils/mstpctl-utils-functions.sh]: yep ; shellcheck will not complain if i added the 2 dashes ; wasn't sure, so i just silenced shellcheck

File: utils/mstpctl-utils-functions.sh (hunk 1 above)
Comment (commodo) [inline utils/mstpctl-utils-functions.sh]: ummm , i'm unclear about what you mean by "we are forking the development of this file"

Comment (PaulSD) [inline utils/mstpctl-utils-functions.sh]: I mean: If Debian makes changes or bug fixes to that file, we will no longer be able to simply `cp /lib/bridge-utils/bridge-utils.sh utils/mstpctl-utils-functions.sh` to update our version with their changes.
Not a big deal, just something to be aware of.

**Comment by commodo** [comment]
thanks @PaulSD for reviewing :)
anyone have any other comments ?

**Comment by vivien** [comment]
Looks good to me to. @commodo are you going to send a new PR or push-force this one, or is it already ready to merge as-is?

**Comment by commodo** [comment]
i already force pushed on the other 2 comments
i double checked and it seems fine

**Comment by vivien** [comment]
Thanks guys!

**Pull request merged**